src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs

```diff
@@ -0,0 +1,479 @@
+        {
+            var resolved = await Dns.GetHostEntryAsync(address.Host);
+            var found = resolved.AddressList.FirstOrDefault(a => a.AddressFamily == addressFamily);
+            if (found == null)
+            {
+                throw new KeyNotFoundException($"Couldn't resolve IP endpoint from provided DNS name '{address}' with address family of '{addressFamily}'");
+            }
+
+            return new IPEndPoint(found, address.Port);
+        }
+
+        #endregion
+
+        #region static methods
+
+        public static Address MapSocketToAddress(IPEndPoint socketAddress, string schemeIdentifier, string systemName, string hostName = null)
+        {
+            return socketAddress == null
+                ? null
+                : new Address(schemeIdentifier, systemName, SafeMapHostName(hostName) ?? SafeMapIPv6(socketAddress.Address), socketAddress.Port);
+        }
+
+        private static string SafeMapHostName(string hostName)
+        {
+            IPAddress ip;
+            return !string.IsNullOrEmpty(hostName) && IPAddress.TryParse(hostName, out ip) ? SafeMapIPv6(ip) : hostName;
+        }
+
+        private static string SafeMapIPv6(IPAddress ip) => ip.AddressFamily == AddressFamily.InterNetworkV6 ? "[" + ip + "]" : ip.ToString();
+
+        public static EndPoint ToEndpoint(Address address)
+        {
+            if (!address.Port.HasValue) throw new ArgumentNullException(nameof(address), $"Address port must not be null: {address}");
+
+            IPAddress ip;
+            return IPAddress.TryParse(address.Host, out ip)
+                ? (EndPoint)new IPEndPoint(ip, address.Port.Value)
+                : new DnsEndPoint(address.Host, address.Port.Value);
+        }
+
+        /// <summary>
+        /// Maps an Akka.NET address to correlated <see cref="EndPoint"/>.
+        /// </summary>
+        /// <param name="address">Akka.NET fully qualified node address.</param>
+        /// <exception cref="ArgumentException">Thrown if address port was not provided.</exception>
+        /// <returns><see cref="IPEndPoint"/> for IP-based addresses, <see cref="DnsEndPoint"/> for named addresses.</returns>
+        public static EndPoint AddressToSocketAddress(Address address)
+        {
+            if (address.Port == null) throw new ArgumentException($"address port must not be null: {address}");
+            EndPoint listenAddress;
+            IPAddress ip;
+            if (IPAddress.TryParse(address.Host, out ip))
+            {
+                listenAddress = new IPEndPoint(ip, (int)address.Port);
+            }
+            else
+            {
+                // DNS resolution will be performed by the transport
+                listenAddress = new DnsEndPoint(address.Host, (int)address.Port);
+            }
+            return listenAddress;
+        }
+
+        #endregion
+    }
+
+    internal class HeliosBackwardsCompatabilityLengthFramePrepender : LengthFieldPrepender
```

Comment:
Yes. I've taken it from there + added ByteOrder.LittleEndian (because of DotNetty :) )